sirepo/package_data/static/js/sirepo-plotting.js

```diff
@@ -3052,992 +3071,110 @@ SIREPO.app.directive('particle', function(plotting, layoutService, utilities) {
-                var osBottom = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1] + 0.5 * outlineSource.getYLength(), osCenter[2] + 0.5 * outlineSource.getZLength()];
-                var osBottomIn = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1] + 0.5 * outlineSource.getYLength(), osCenter[2] - 0.5 * outlineSource.getZLength()];
-                var osTop = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1] - 0.5 * outlineSource.getYLength(), osCenter[2] + 0.5 * outlineSource.getZLength()];
-                var worldCoord = vtk.Rendering.Core.vtkCoordinate.newInstance({
-                    renderer: renderer
-                });
-                worldCoord.setCoordinateSystemToWorld();
-
-                // viewport values seem to be double, not sure why
-                worldCoord.setValue(osLeft);
-                var vpLeft = worldCoord.getComputedViewportValue();
-                var vpLeftX = vpLeft[0] / 2.0;
-
-                worldCoord.setValue(osLeftIn);
-                var vpLeftIn = worldCoord.getComputedViewportValue()[0] / 2.0;
-
-                worldCoord.setValue(osRight);
-                var vpRight = worldCoord.getComputedViewportValue();
-                var vpRightX = vpRight[0] / 2.0;
-
-                worldCoord.setValue(osBottom);
-                var vpBottom = worldCoord.getComputedViewportValue();
-                var vpBottomY = vpBottom[1] / 2.0;
-
-                worldCoord.setValue(osBottomIn);
-                var vpBottomIn = worldCoord.getComputedViewportValue();
-
-                worldCoord.setValue(osTop);
-                var vpTop = worldCoord.getComputedViewportValue();
-                var vpTopY = vpTop[1] / 2.0;
-
-                var vpWidth = Math.abs(vpRightX - vpLeftX);
-                var vpHeight = Math.abs(vpTopY - vpBottomY);
-                var tanTheta = (vpBottom[1] - vpBottomIn[1]) / (vpBottom[0] - vpBottomIn[0]);
-                var theta = 180 * Math.atan(tanTheta) / Math.PI;
-                $scope.zAxisAngle = theta;
+    this.isPointWithinBounds = function (p, b) {
+        return p[0] >= b.left && p[0] <= b.right && p[1] >= b.top && p[1] <= b.bottom;
+    };
 
-                // little test boxes are useful for translating vtk space to screen space
-                $scope.testBoxes = [
-                    {
-                        x: vpTop[0] / 2.0,
-                        y: vpTop[1] / 2.0,
-                        color: "red"
-                    },
-                    {
-                        x: vpBottom[0] / 2.0,
-                        y: vpBottom[1] / 2.0,
-                        color: "blue"
-                    }
-                ];
+    // Returns the point(s) that have the smallest (reverse == false) or largest value in the given dimension
+    this.extrema = function(pArr, dim, reverse) {
+        var sPArr = self.sortInDimension(pArr, dim, reverse);
```

Comment:
I'd just call it "res" or "result" not sPArr. It's just the value cascading through the function. It lets you know that when you say "result" you are operating on the return value right from the start, just performing different ops along the way.